### PR TITLE
modules: hal_nordic: Remove nrf_802154_lp_timer platform

### PR DESCRIPTION
**Pull request by e-rk**
The nrf_802154_lp_timer_zephyr.c platform file is unused in the
opensource SL library variant.

Signed-off-by: Rafał Kuźnia <rafal.kuznia@nordicsemi.no>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

File: modules/hal_nordic/nrf_802154/CMakeLists.txt [hunk 1]
```diff
@@ -98,7 +91,6 @@ else()
   target_compile_definitions(zephyr-802154-interface INTERFACE NRF_802154_INTERNAL_RADIO_IRQ_HANDLING=0)
 endif()
 
-set(NRF52_SERIES  ${CONFIG_SOC_SERIES_NRF52X})
```
Comment (anangl) [inline modules/hal_nordic/nrf_802154/CMakeLists.txt:101]: This change does not seem to be related. It would be better to move it to a separate commit.

Comment (e-rk) [reply modules/hal_nordic/nrf_802154/CMakeLists.txt]: Removed the change from PR because it is not related to the PR topic.

**Review by rlubos — APPROVED** [review]

**Review by anangl — APPROVED** [review]
